src/compiler/rust/canister/src/lib.rs

```diff
@@ -214,11 +228,60 @@ fn run_event_loop(context: &mut wasmedge_quickjs::Context) {
 #[ic_cdk_macros::update]
 pub fn _azle_chunk() {}
 
-fn get_consumer(consumer_path: &str) -> Result<Consumer, String> {
-    let consumer_string = std::fs::read_to_string(consumer_path)
-        .map_err(|err| format!("Error reading {consumer_path}: {err}"))?;
-    let consumer: Consumer = serde_json::from_str(&consumer_string)
-        .map_err(|err| format!("Error parsing {consumer_path}: {err}"))?;
+#[inline(never)]
+#[no_mangle]
+extern "C" fn init_js_passive_data(js_vec_location: i32) -> usize {
+    "123_456_789".parse::<usize>().unwrap() + js_vec_location as usize // TODO must be like this for weird optimization reasons
+}
+
+// TODO seems we need to do this to stop the compiler from hard-coding the result of this function where it is called
+// TODO hopefully there's a less hacky way to do this
+#[inline(never)]
+#[no_mangle]
+extern "C" fn js_passive_data_size() -> usize {
+    "123_456_789".parse().unwrap()
+}
+
+// TODO waiting on license inspired from https://github.com/adambratschikaye/wasm-inject-data/blob/main/src/static_wasm.rs
+#[inline(never)]
+#[no_mangle]
+pub extern "C" fn get_js_code() -> Vec<u8> {
+    let size = unsafe { js_passive_data_size() };
+    let mut js_vec = vec![243; size];
+    let js_vec_location = js_vec.as_mut_ptr() as i32;
+
+    unsafe {
+        init_js_passive_data(js_vec_location);
+    }
+
+    js_vec
+}
+
+#[inline(never)]
+#[no_mangle]
+extern "C" fn init_wasm_data_passive_data(js_vec_location: i32) -> usize {
```

Comment:
hey this should be called wasm_vec_location

Reply:
Changed